helix-view/src/editor.rs

```diff
@@ -755,8 +764,9 @@ impl Editor {
         Ok(self.new_file_from_document(action, Document::from(rope, Some(encoding))))
     }
 
-    pub fn open(&mut self, path: PathBuf, action: Action) -> Result<DocumentId, Error> {
-        let path = helix_core::path::get_canonicalized_path(&path)?;
+    // ??? possible use for integration tests
+    pub fn open(&mut self, path: &Path, action: Action) -> Result<DocumentId, Error> {
+        let path = helix_core::path::get_canonicalized_path(path)?;
```

Comment:
Do we want to change this? Why not do `get_canonicalized_path` outside so we don't have to allocate again inside this function if possible?

Reply:
I think doing it here is desirable. It's defensively making sure the path is right. If we just assume that the input is always good, then we will get crashes. Also this is just the open command. It is not in a critical path where we need to worry about unnecessary allocations.